### PR TITLE
composer release 1.8.5

### PR DESCRIPTION
**Pull request by alcohol**

### CODE REVIEW

File: library/composer [hunk 1]
```diff
@@ -3,12 +3,12 @@
 Maintainers: Composer (@composer), Rob Bast (@alcohol)
 GitRepo: https://github.com/composer/docker.git
 
-Tags: 1.8.4, 1.8, 1, latest
-Architectures: amd64, arm32v6, arm64v8, i386, ppc64le
-GitCommit: ebbb9efd87b78cf9b837f97105bcb6c5f0ee44ef
+Tags: 1.8.5, 1.8, 1, latest
+Architectures: amd64, arm32v6, arm32v7, arm64v8, i386, ppc64le
+GitCommit: be36910e4d2a3b5a055d2075982e04c835c5db2e
```
Comment (tianon) [inline library/composer]: This commit doesn't appear to be part of any branches on the repo -- is it the right one?

Comment (alcohol) [reply library/composer]: Let me investigate what went wrong here.

Comment (alcohol) [reply library/composer]: Sorry, I merged my branch on Github (using rebase) but forget to checkout master on my local machine. It was a late night :sweat:

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index c98f2aa..bf59b52 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,10 +1,12 @@
 composer:1.7 @ amd64
 composer:1.7 @ arm32v6
+composer:1.7 @ arm32v7
 composer:1.7 @ arm64v8
 composer:1.7 @ i386
 composer:1.7 @ ppc64le
 composer:latest @ amd64
 composer:latest @ arm32v6
+composer:latest @ arm32v7
 composer:latest @ arm64v8
 composer:latest @ i386
 composer:latest @ ppc64le
diff --git a/_bashbrew-list b/_bashbrew-list
index 22e1648..8de451c 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -2,5 +2,5 @@ composer:1
 composer:1.7
 composer:1.7.3
 composer:1.8
-composer:1.8.4
+composer:1.8.5
 composer:latest
diff --git a/composer_1.7/Dockerfile b/composer_1.7/Dockerfile
index 42e7eb5..6c84b28 100644
--- a/composer_1.7/Dockerfile
+++ b/composer_1.7/Dockerfile
@@ -1,21 +1,22 @@
-FROM php:7-alpine3.8
+FROM php:7-alpine
 
-RUN apk --no-cache add --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip
-
-RUN echo "memory_limit=-1" > "$PHP_INI_DIR/conf.d/memory-limit.ini" \
- && echo "date.timezone=${PHP_TIMEZONE:-UTC}" > "$PHP_INI_DIR/conf.d/date_timezone.ini"
-
-RUN apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
+RUN apk add --upgrade --no-cache --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip coreutils \
+ && apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
  && docker-php-ext-configure zip --with-libzip \
- && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip \
+ && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip opcache \
  && runDeps="$( \
     scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
       | tr ',' '\n' \
       | sort -u \
       | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
     )" \
- && apk add --virtual .composer-phpext-rundeps $runDeps \
- && apk del .build-deps
+ && apk add --no-cache --virtual .composer-phpext-rundeps $runDeps \
+ && apk del .build-deps \
+ && printf "# composer php cli ini settings\n\
+date.timezone=UTC\n\
+memory_limit=-1\n\
+opcache.enable_cli=1\n\
+" > $PHP_INI_DIR/php-cli.ini
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
 ENV COMPOSER_HOME /tmp
diff --git a/composer_latest/Dockerfile b/composer_latest/Dockerfile
index 9de8fa4..564fd8d 100644
--- a/composer_latest/Dockerfile
+++ b/composer_latest/Dockerfile
@@ -1,25 +1,26 @@
-FROM php:7-alpine3.8
+FROM php:7-alpine
 
-RUN apk --no-cache add --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip
-
-RUN echo "memory_limit=-1" > "$PHP_INI_DIR/conf.d/memory-limit.ini" \
- && echo "date.timezone=${PHP_TIMEZONE:-UTC}" > "$PHP_INI_DIR/conf.d/date_timezone.ini"
-
-RUN apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
+RUN apk add --upgrade --no-cache --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip coreutils \
+ && apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
  && docker-php-ext-configure zip --with-libzip \
- && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip \
+ && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip opcache \
  && runDeps="$( \
     scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
       | tr ',' '\n' \
       | sort -u \
       | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
     )" \
- && apk add --virtual .composer-phpext-rundeps $runDeps \
- && apk del .build-deps
+ && apk add --no-cache --virtual .composer-phpext-rundeps $runDeps \
+ && apk del .build-deps \
+ && printf "# composer php cli ini settings\n\
+date.timezone=UTC\n\
+memory_limit=-1\n\
+opcache.enable_cli=1\n\
+" > $PHP_INI_DIR/php-cli.ini
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
 ENV COMPOSER_HOME /tmp
-ENV COMPOSER_VERSION 1.8.4
+ENV COMPOSER_VERSION 1.8.5
 
 RUN curl --silent --fail --location --retry 3 --output /tmp/installer.php --url https://raw.githubusercontent.com/composer/getcomposer.org/cb19f2aa3aeaa2006c0cd69a7ef011eb31463067/web/installer \
  && php -r " \
```

</details>

**Comment by yosifkit** [comment]
```diff
-FROM php:7-alpine3.8
+FROM php:7-alpine
```

This may run hit possible build issues in the future when the php image moves to the next alpine release. Not an issue right now, but could cause problems later.

>  apk add --upgrade

`upgrade` is often going to cause problems -- we prefer to instead update the base image and rebuild

> Avoid `RUN apt-get upgrade` and `dist-upgrade`, as many of the “essential” packages from the parent images cannot upgrade inside an unprivileged container. If a package contained in the parent image is out-of-date, contact its maintainers.
>
> \- https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

**Comment by alcohol** [comment]
I'm not sure how Alpine versions could ever cause issues for Composer, provided the PHP images still delivers a working PHP version.

If `--upgrade` is really so frowned upon, I guess I will remove it. :man_shrugging:

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index c98f2aa..bf59b52 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,10 +1,12 @@
 composer:1.7 @ amd64
 composer:1.7 @ arm32v6
+composer:1.7 @ arm32v7
 composer:1.7 @ arm64v8
 composer:1.7 @ i386
 composer:1.7 @ ppc64le
 composer:latest @ amd64
 composer:latest @ arm32v6
+composer:latest @ arm32v7
 composer:latest @ arm64v8
 composer:latest @ i386
 composer:latest @ ppc64le
diff --git a/_bashbrew-list b/_bashbrew-list
index 22e1648..8de451c 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -2,5 +2,5 @@ composer:1
 composer:1.7
 composer:1.7.3
 composer:1.8
-composer:1.8.4
+composer:1.8.5
 composer:latest
diff --git a/composer_1.7/Dockerfile b/composer_1.7/Dockerfile
index 42e7eb5..0f7ace0 100644
--- a/composer_1.7/Dockerfile
+++ b/composer_1.7/Dockerfile
@@ -1,21 +1,22 @@
-FROM php:7-alpine3.8
+FROM php:7-alpine
 
-RUN apk --no-cache add --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip
-
-RUN echo "memory_limit=-1" > "$PHP_INI_DIR/conf.d/memory-limit.ini" \
- && echo "date.timezone=${PHP_TIMEZONE:-UTC}" > "$PHP_INI_DIR/conf.d/date_timezone.ini"
-
-RUN apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
+RUN apk add --no-cache --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip coreutils \
+ && apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
  && docker-php-ext-configure zip --with-libzip \
- && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip \
+ && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip opcache \
  && runDeps="$( \
     scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
       | tr ',' '\n' \
       | sort -u \
       | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
     )" \
- && apk add --virtual .composer-phpext-rundeps $runDeps \
- && apk del .build-deps
+ && apk add --no-cache --virtual .composer-phpext-rundeps $runDeps \
+ && apk del .build-deps \
+ && printf "# composer php cli ini settings\n\
+date.timezone=UTC\n\
+memory_limit=-1\n\
+opcache.enable_cli=1\n\
+" > $PHP_INI_DIR/php-cli.ini
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
 ENV COMPOSER_HOME /tmp
diff --git a/composer_latest/Dockerfile b/composer_latest/Dockerfile
index 9de8fa4..dfbe746 100644
--- a/composer_latest/Dockerfile
+++ b/composer_latest/Dockerfile
@@ -1,25 +1,26 @@
-FROM php:7-alpine3.8
+FROM php:7-alpine
 
-RUN apk --no-cache add --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip
-
-RUN echo "memory_limit=-1" > "$PHP_INI_DIR/conf.d/memory-limit.ini" \
- && echo "date.timezone=${PHP_TIMEZONE:-UTC}" > "$PHP_INI_DIR/conf.d/date_timezone.ini"
-
-RUN apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
+RUN apk add --no-cache --virtual .composer-rundeps git subversion openssh mercurial tini bash patch make zip unzip coreutils \
+ && apk add --no-cache --virtual .build-deps zlib-dev libzip-dev \
  && docker-php-ext-configure zip --with-libzip \
- && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip \
+ && docker-php-ext-install -j$(getconf _NPROCESSORS_ONLN) zip opcache \
  && runDeps="$( \
     scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
       | tr ',' '\n' \
       | sort -u \
       | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
     )" \
- && apk add --virtual .composer-phpext-rundeps $runDeps \
- && apk del .build-deps
+ && apk add --no-cache --virtual .composer-phpext-rundeps $runDeps \
+ && apk del .build-deps \
+ && printf "# composer php cli ini settings\n\
+date.timezone=UTC\n\
+memory_limit=-1\n\
+opcache.enable_cli=1\n\
+" > $PHP_INI_DIR/php-cli.ini
 
 ENV COMPOSER_ALLOW_SUPERUSER 1
 ENV COMPOSER_HOME /tmp
-ENV COMPOSER_VERSION 1.8.4
+ENV COMPOSER_VERSION 1.8.5
 
 RUN curl --silent --fail --location --retry 3 --output /tmp/installer.php --url https://raw.githubusercontent.com/composer/getcomposer.org/cb19f2aa3aeaa2006c0cd69a7ef011eb31463067/web/installer \
  && php -r " \
````

</details>

**Comment by tianon** [comment]
LGTM

Build test of #5704; 6ad28a88; `amd64` (`composer`):

```console

$ bashbrew build composer:1.8.5
Building bashbrew/cache:e45d7101ad81167776bd03a9f938f6107a825770a607c8f001a393fa148b9e0f (composer:1.8.5)
Tagging composer:1.8.5
Tagging composer:1.8
Tagging composer:1
Tagging composer:latest

$ test/run.sh composer:1.8.5
testing composer:1.8.5
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'composer' [5/5]...Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 16 installs, 0 updates, 0 removals
  - Installing symfony/process (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (15%)Downloading (35%)Downloading (45%)Downloading (60%)Downloading (75%)Downloading (95%)Downloading (100%)
  - Installing symfony/finder (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing symfony/polyfill-ctype (v1.11.0): Downloading (connecting...)Downloading (0%)           Downloading (100%)
  - Installing symfony/filesystem (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (35%)Downloading (70%)Downloading (90%)Downloading (100%)
  - Installing symfony/polyfill-mbstring (v1.11.0): Downloading (connecting...)Downloading (0%)           Downloading (30%)Downloading (55%)Downloading (100%)
  - Installing symfony/contracts (v1.0.2): Downloading (connecting...)Downloading (0%)           Downloading (55%)Downloading (60%)Downloading (75%)Downloading (100%)
  - Installing symfony/console (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (65%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (85%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing psr/log (1.1.0): Downloading (connecting...)Downloading (0%)           Downloading (65%)Downloading (100%)
  - Installing seld/phar-utils (1.0.1): Downloading (connecting...)Downloading (0%)           Downloading (30%)Downloading (60%)Downloading (90%)Downloading (100%)
  - Installing seld/jsonlint (1.7.1): Downloading (connecting...)Downloading (0%)           Downloading (15%)Downloading (75%)Downloading (90%)Downloading (100%)
  - Installing justinrainbow/json-schema (5.2.8): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (20%)Downloading (25%)Downloading (45%)Downloading (50%)Downloading (65%)Downloading (70%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing composer/xdebug-handler (1.3.2): Downloading (connecting...)Downloading (0%)           Downloading (25%)Downloading (55%)Downloading (95%)Downloading (100%)
  - Installing composer/spdx-licenses (1.5.1): Downloading (connecting...)Downloading (0%)           Downloading (10%)Downloading (20%)Downloading (100%)
  - Installing composer/semver (1.5.0): Downloading (connecting...)Downloading (0%)           Downloading (35%)Downloading (85%)Downloading (100%)
  - Installing composer/ca-bundle (1.1.4): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (65%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (85%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing composer/composer (1.8.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (65%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (85%)Downloading (90%)Downloading (95%)Downloading (100%)
symfony/contracts suggests installing psr/cache (When using the Cache contracts)
symfony/contracts suggests installing psr/container (When using the Service contracts)
symfony/contracts suggests installing symfony/cache-contracts-implementation
symfony/contracts suggests installing symfony/service-contracts-implementation
symfony/contracts suggests installing symfony/translation-contracts-implementation
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
Writing lock file
Generating autoload files
passed


$ bashbrew build composer:1.7.3
Building bashbrew/cache:62d22a032b9a367761d2577fd0291d0cc902b0e17eb2f44699e74862f586495f (composer:1.7.3)
Tagging composer:1.7.3
Tagging composer:1.7

$ test/run.sh composer:1.7.3
testing composer:1.7.3
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'composer' [5/5]...Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 16 installs, 0 updates, 0 removals
  - Installing symfony/process (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (15%)Downloading (35%)Downloading (45%)Downloading (60%)Downloading (75%)Downloading (95%)Downloading (100%)
  - Installing symfony/finder (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing symfony/polyfill-ctype (v1.11.0): Downloading (connecting...)Downloading (0%)           Downloading (100%)
  - Installing symfony/filesystem (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (35%)Downloading (70%)Downloading (100%)
  - Installing symfony/polyfill-mbstring (v1.11.0): Downloading (connecting...)Downloading (0%)           Downloading (30%)Downloading (55%)Downloading (85%)Downloading (100%)
  - Installing symfony/contracts (v1.0.2): Downloading (connecting...)Downloading (0%)           Downloading (45%)Downloading (75%)Downloading (100%)
  - Installing symfony/console (v4.2.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (65%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (85%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing psr/log (1.1.0): Downloading (connecting...)Downloading (0%)           Downloading (65%)Downloading (90%)Downloading (100%)
  - Installing seld/phar-utils (1.0.1): Downloading (connecting...)Downloading (0%)           Downloading (30%)Downloading (60%)Downloading (90%)Downloading (100%)
  - Installing seld/jsonlint (1.7.1): Downloading (connecting...)Downloading (0%)           Downloading (15%)Downloading (75%)Downloading (90%)Downloading (100%)
  - Installing justinrainbow/json-schema (5.2.8): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (20%)Downloading (25%)Downloading (40%)Downloading (45%)Downloading (65%)Downloading (70%)Downloading (85%)Downloading (90%)Downloading (100%)
  - Installing composer/xdebug-handler (1.3.2): Downloading (connecting...)Downloading (0%)           Downloading (25%)Downloading (55%)Downloading (95%)Downloading (100%)
  - Installing composer/spdx-licenses (1.5.1): Downloading (connecting...)Downloading (0%)           Downloading (10%)Downloading (20%)Downloading (100%)
  - Installing composer/semver (1.5.0): Downloading (connecting...)Downloading (0%)           Downloading (35%)Downloading (85%)Downloading (100%)
  - Installing composer/ca-bundle (1.1.4): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (65%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (85%)Downloading (90%)Downloading (95%)Downloading (100%)
  - Installing composer/composer (1.8.5): Downloading (connecting...)Downloading (0%)           Downloading (5%)Downloading (10%)Downloading (15%)Downloading (20%)Downloading (25%)Downloading (30%)Downloading (35%)Downloading (40%)Downloading (45%)Downloading (50%)Downloading (55%)Downloading (60%)Downloading (70%)Downloading (75%)Downloading (80%)Downloading (90%)Downloading (95%)Downloading (100%)
symfony/contracts suggests installing psr/cache (When using the Cache contracts)
symfony/contracts suggests installing psr/container (When using the Service contracts)
symfony/contracts suggests installing symfony/cache-contracts-implementation
symfony/contracts suggests installing symfony/service-contracts-implementation
symfony/contracts suggests installing symfony/translation-contracts-implementation
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
Writing lock file
Generating autoload files
passed

```

**Pull request merged**